File: Leetcode/605. Can Place Flowers/solution.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        size = len(flowerbed)
        count = 0

        for i, flower in enumerate(flowerbed):

            if not flower and (not i or not flowerbed[i-1]) and (i == size-1 or not flowerbed[i+1]):
                count += 1
                flowerbed[i] = 1

            if count >= n:
                return True

        return False

File: Leetcode/605. Can Place Flowers/solution.rs
impl Solution {
    pub fn can_place_flowers(flowerbed: Vec<i32>, n: i32) -> bool {
        let mut count = 0;
        let mut size = flowerbed.len() - 1;
        let mut flowerbed = flowerbed.clone();
        
        for i in 0..flowerbed.len() {
            if flowerbed[i] == 0 && (i == 0 || flowerbed[i-1] == 0) && (i == size || flowerbed[i+1] == 0) {
                flowerbed[i] = 1;
                count += 1;
            }
            
            if count >= n {
                return true;
            }
        }
        
        false
    }
}